src/RestaurantsContainer.jsx

```diff
@@ -1,17 +1,34 @@
 import React from 'react';
 
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  Link,
+} from 'react-router-dom';
 
 import { get } from './utils';
+import { loadRestaurantDetail } from './actions';
 
 export default function RestaurantsContainer() {
+  const dispatch = useDispatch();
   const restaurants = useSelector(get('restaurants'));
 
+  function handleClick(restaurantId) {
+    dispatch(loadRestaurantDetail(restaurantId));
+  }
+
   return (
     <ul>
       {restaurants.map((restaurant) => (
```

Comment:
구조분해할당을 적용해도 좋겠네요 ~

Reply:
```js
this.state = {
  id: 1,
  name: 'kim',
}

const { id, name } = this.state;
```

이 경우에 적용하는게 구조분해할당이라고 알고있어요.

handleClick, dispatch에 구조분해할당을 어떻게 하는지 감이 잘 안오네요 ..

힌트를 더 주실 수 있으신가요 ?

Reply:
> ```js
> this.state = {
>   id: 1,
>   name: 'kim',
> }
> 
> const { id, name } = this.state;
> ```
> 
> 이 경우에 적용하는게 구조분해할당이라고 알고있어요.
> 
> handleClick, dispatch에 구조분해할당을 어떻게 하는지 감이 잘 안오네요 ..
> 
> 힌트를 더 주실 수 있으신가요 ?

넵 제가 설명이 조금 부족했네요😅 알고 계신 부분을 응용한다고 생각하시면 좋을 것 같아요. 

지금은 map 내부에서 id와 name 만 사용하고 있으니 이렇게 구조분해할당해서 바로 쓸 수 있겠네요 ㅎㅎ 

```suggestion
      {restaurants.map(({ id, name }) => (
```

- https://ko.javascript.info/destructuring-assignment#ref-673

Reply:
오 .. 처음 state나 다른 객체로부터 받을 때만 적용할 수 있다고 생각이 갇혀있었어요.

통으로 받고나서 반복문 내부에서 호출할 때도 이렇게 활용 할 수 있겠네요.

좋은 팁 감사합니다 ~~